src/services/talkService.ts

```diff
@@ -106,61 +164,288 @@ export async function createTalkRoomFromProposal(proposal: ProposalInterface): P
+			.map(async (result) => {
+				try {
+					if (result.userId) {
+						await transceivePost<TalkRoomAddParticipantRequest, { type: number }>(
+							`room/${token}/participants`,
+							{
+								newParticipant: result.userId,
+								source: 'users',
+							},
+						)
+						logger.debug('Added user participant', { userId: result.userId })
+					} else if (result.email) {
+						await transceivePost<TalkRoomAddParticipantRequest, { type: number }>(
+							`room/${token}/participants`,
+							{
+								newParticipant: result.email,
+								source: 'emails',
+							},
+						)
+						logger.debug('Added email participant', { email: result.email })
+					}
+				} catch (error) {
+					logger.info('Could not add participant', {
+						userId: result.userId,
+						email: result.email,
+						error,
+					})
+				}
+			})
+
+		await Promise.all(updatePromises)
+		logger.debug('Finished updating room participants')
+	} catch (error) {
+		logger.warn('Could not update Talk room attendees', { error })
+	}
+}
+
+/**
+ * Generic function to GET data from the Talk OCS API and return the unwrapped data
+ *
+ * @param path API path segment to append after the version (for example: "room")
+ * @param params The request query parameters to append to the URL
+ *
+ * @return Promise that resolves to the unwrapped OCS data
+ *
+ * @throws Error if the request fails or OCS response is invalid
+ */
+async function transceiveGet<TRequest extends object | undefined, TResponse>(path: string, params?: TRequest): Promise<TResponse> {
+	const apiVersion = loadState('calendar', 'talk_api_version')
+	const url = generateOcsUrl('/apps/spreed/api/{apiVersion}/{path}', { apiVersion, path })
+
+	let response: AxiosResponse<TResponse>
+	try {
+		response = await axios.get<TResponse>(url, {
+			params,
+			headers: {
+				'OCS-APIREQUEST': 'true',
+			},
+		})
+		// Check if response has OCS envelope structure
+		if (response.data && typeof response.data === 'object' && 'ocs' in response.data) {
+			const ocsResponse = response.data as OcsEnvelope<TResponse>
+			// Response sanity checks
+			if (!ocsResponse.ocs || !ocsResponse.ocs.meta || !ocsResponse.ocs.data || !ocsResponse.ocs.meta.status) {
+				throw new Error('Talk service error: malformed response')
+			}
+			if (ocsResponse.ocs.meta.status !== 'ok' && ocsResponse.ocs.meta.message) {
+				throw new Error(`Talk service error: ${ocsResponse.ocs.meta.message}`)
+			}
+			if (ocsResponse.ocs.meta.status !== 'ok' && typeof ocsResponse.ocs.data === 'object' && 'error' in ocsResponse.ocs.data) {
+				throw new Error(`Talk service error: ${ocsResponse.ocs.data.error}`)
+			}
+			if (ocsResponse.ocs.meta.status !== 'ok') {
+				throw new Error('Talk service error: unknown error')
+			}
+			return ocsResponse.ocs.data as TResponse
+		}
+		// If not an OCS envelope, return the data as-is
+		return response.data as TResponse
 	} catch (error) {
-		console.debug(error)
```

Comment:
This is important. You are removing the debuggability from this code. Bring logging back!